aws/rust-runtime/aws-inlineable/src/presigning.rs

```diff
@@ -168,62 +170,46 @@ impl PresigningConfigBuilder {
 ///
 /// **This struct has conversion convenience functions:**
 ///
-/// - [`PresignedRequest::to_http_request<B>`][Self::to_http_request] returns an [`http::Request<B>`](https://docs.rs/http/0.2.6/http/request/struct.Request.html)
+/// - [`PresignedRequest::to_http_02x_request<B>`][Self::to_http_03x_request] returns an [`http::Request<B>`](https://docs.rs/http/0.2.6/http/request/struct.Request.html)
 /// - [`PresignedRequest::into`](#impl-From<PresignedRequest>) returns an [`http::request::Builder`](https://docs.rs/http/0.2.6/http/request/struct.Builder.html)
 #[non_exhaustive]
-pub struct PresignedRequest(http::Request<()>);
+pub struct PresignedRequest(HttpRequest);
 
 impl PresignedRequest {
     #[allow(dead_code)]
-    pub(crate) fn new(inner: http::Request<()>) -> Self {
+    pub(crate) fn new(inner: HttpRequest) -> Self {
         Self(inner)
     }
 
     /// Returns the HTTP request method.
-    pub fn method(&self) -> &http::Method {
+    pub fn method(&self) -> &str {
         self.0.method()
     }
 
     /// Returns the HTTP request URI.
-    pub fn uri(&self) -> &http::Uri {
+    pub fn uri(&self) -> &str {
         self.0.uri()
     }
 
     /// Returns any HTTP headers that need to go along with the request, except for `Host`,
     /// which should be sent based on the endpoint in the URI by the HTTP client rather than
     /// added directly.
-    pub fn headers(&self) -> &http::HeaderMap<http::HeaderValue> {
-        self.0.headers()
+    pub fn headers(&self) -> impl Iterator<Item = (&str, &str)> {
```

Comment:
Do we need the sensitive flag on headers for presigning?

Reply:
my take is this header is probably not _that_ sensitive to make it worthwhile to return a custom type here. Open to thoughts though.